src/ProjectTemplates/BlazorWasm.ProjectTemplates/content/BlazorWasm-CSharp/Client/Program.cs

```diff
@@ -8,13 +13,19 @@ namespace BlazorWasm_CSharp
 {
     public class Program
     {
-        public static void Main(string[] args)
+        public static async Task Main(string[] args)
         {
-            CreateHostBuilder(args).Build().Run();
-        }
+            var builder = WebAssemblyHostBuilder.CreateDefault(args);
+            builder.RootComponents.Add<App>("app");
 
-        public static IWebAssemblyHostBuilder CreateHostBuilder(string[] args) =>
-            BlazorWebAssemblyHost.CreateDefaultBuilder()
-                .UseBlazorStartup<Startup>();
+            // use builder.Services to configure application services.
+#if (IndividualLocalAuth)
+            builder.Services.AddOptions();
```

Comment:
@SteveSandersonMS @danroth27 - thoughts about this?  https://github.com/dotnet/aspnetcore/issues/18471

Someone has to register options in DI. Doing it in Blazor's host means everyone gets it regardless of whether they need it or not. We can fix the underlying issue in 5.0.

Reply:
@rynowak does builder.Services.AddAuthorizationCore not call AddOptions internally?

Is that something we could do for you in WebAssemblyHostBuilder.CreateDefault?

Normally the `CreateDefault` method is where we put all those type of opinions.

Reply:
I see now that you want to drop options from the default builder. I think that's fine, I'm just wondering if any realistic app would end up using it either-way and at that point what value there is for removing it for the 10-20% people don't use it.

Reply:
> I see now that you want to drop options from the default builder. I think that's fine, I'm just wondering if any realistic app would end up using it either-way and at that point what value there is for removing it for the 10-20% people don't use it.

We can definitely hide the call, we hid it before. That increases our size. 

I would wager it goes the other way - the apps that use auth will use options. If you don't use auth you won't use is. So I'm not sure 20% is the right way to think about it.

Reply:
Seems OK to me to have it like this for now. Like you say we can make it more streamlined in 5.0.